torch/ao/nn/qat/dynamic/modules/linear.py

```diff
@@ -20,15 +22,15 @@ class Linear(torch.ao.nn.qat.Linear):
 
     def __init__(
         self,
-        in_features,
-        out_features,
-        bias=True,
-        qconfig=None,
-        device=None,
-        dtype=None,
+        in_features: int,
+        out_features: int,
+        bias: bool = True,
+        qconfig: Optional[QConfig] = None,
+        device: Optional[Union[int, str, torch.device]] = None,
+        dtype: Optional[str] = None,
     ) -> None:
         super().__init__(in_features, out_features, bias, qconfig, device, dtype)
-        if not torch.ao.quantization.qconfig._activation_is_memoryless(qconfig):
+        if not torch.ao.quantization.qconfig._activation_is_memoryless(qconfig):  # type: ignore[arg-type]
```

Comment:
needs a None check? or use the `not_none` utility?

Reply:
I worry about adding runtime changes (since it can effect correctness + perf) when adding typing for files I'm not too familiar with. I suppose this should be relatively risk-free though...